# Parameters for our MPC Controller Node
mpc_controller:
  ros__parameters:
    # Longer prediction horizon to see further ahead
    P: 20    # Increased from 15 to 30 steps (6 seconds at Ts=0.2)
    Ts: 0.1  

    # [Δu, Δv, Δr, Δx, Δy, Δψ, x, y, ψ]
    Q_weights: [
      1.0, 1.0, 1.0,    # Velocity changes - high to penalize speed
      100.0, 100.0, 100.0,         # Position changes - zeroed out to simplify cost
      10.0, 10.0, 30.0  # Absolute position error - high for tracking
    ]

    # Higher R weights for smoother control
    R_weights: [1.0, 1.0, 1.0]

    tau_max: [1000.0, 1000.0, 1800.0]
    tau_min: [-1000.0, -1000.0, -1800.0]